### ISSUE TITLE
[NEW]: Add extended issue templates

### ISSUE THREAD

**Issue by yuri-norwood** [yuri-norwood/dotfiles#381]
**Configuration for:**
GitHub

**Files required:**
* `.github/`
  * `ISSUE_TEMPLATE/`
    * [x] `vim-colourscheme.md` #384
      * For adding or modifying vim colourschemes or changing the `:colo <foo>` invocation
      * Provide colourscheme name and repo link
    * [x] `vim-plugin.md` #385
      * For adding or configuring vim plugins
      * Provide plugin link and description
    * [x] `suckless.md` #386
      * For adding or modifying suckless patches / colourschemes / configs
      * Provide link / rational / palette
    * [ ] `github.md`
      * For adding or modifying GitHub configurations, adding templates, actions, documentation, etc
      * Describe change(s)
    * [ ] `git.md`
      * For adding / modifying `git(1)` configurations, git ignore files, modules, file structure, etc
  * `workflows/`
    * [ ] `pull_request_automation.yml`
      * Update `pr-lint-action` automation to accept new new PR prefixes
        * Note: possible alternative to change all prefixes to `[DOT-xxx]: Foo bar baz`

**Comment by yuri-norwood** [comment]
In the interest of making all templates self consistent, and the PR linter future proof against any number of new issue templates, it may be best have a sing master template that all derive from, and change the prefix of all of them to `DOT`. This will allow a GitHub project titles DOT to be used in the future, as well as limit the complexity of the PR lint regex.

The master template should be usable in the issue creator when no template fits well, this will avoid the current state of similar issues using different templates. The templates themselves should be used for three differentiating purposes:

- Customisation of labels
- Customisation of input fields
- Customisation of title format

**Comment by yuri-norwood** [comment]
Ok, time for a revisit/revamp. This task is getting stale because there isn't a clear idea of what issues for a  dotfiles repo should look like. There are a several goals that the issues should address, some of which overlap with each other, making consistency and clarity tricky:

* Colours
  * Adding / changing colourschemes for vim/nvim
  * Adding / changing colourschemes for suckless tools
  * Adding / changing colourschemes for other tools, e.g. pfetch, xterm, etc
*  Shell environment
  * Startup behavior, e.g. `.xinitrc`
  * Default tools, e.g. `VISUAL` or `EDITOR`
  * Environment variables (may be part of other issues, e.g. changing `GOPATH`)
* GitHub
  * Investigation of actions /  workflows
  * Adding or fixing said actions or workflows (potentially a bug issue type?)
  * Fixing `.gitattribute` or `.gitmodule` (somehow these don't feel like they should be bug issues?)
  * Adding documentation / community files, e.g. README.md, issue templates, sponsor definitions, etc
  * Notes about changing non-code aspects of the repository, labels, tags, releases, etc
* XDG
  * Move or modify existing configs to conform to the XDG base directory spec
  * Add new configs to XDG alternative dirs instead of adding useless/unwanted configs to .gitignore
* Visual Studio changes
  * Basically just toying with `.editorconfig`
* Legit editorconfig changes
  * Fixing file types in specific subdirs where a non-standard config file extension may interfere with linting automation
  *  Adding new filetypes to the main editor config, such as when introducing a new language or a file with semantic whitespace.
* Vim
  * Adding plugins
  * Configuring plugins
  * Fixing a config issue, e.g. beeping or blinking (possibly bug)
  * Adding colourschemes (possibly colours)
  * vim/nvim compatibility fixes

**Comment by yuri-norwood** [comment]
Possible structure:

| Code | Type | Labels | Description |
| --- | --- | --- | --- |
| VIM | Vim/Nvim |  ![vim](https://img.shields.io/github/labels/yuri-norwood/dotfiles/vim) ![nvim](https://img.shields.io/github/labels/yuri-norwood/dotfiles/nvim) | `VIM` describes issues that relate *primarily* to Vim or Neovim, such as maintaining compatibility between the two, managing colourschemes or plugins, etc. |
| XDG | XDG Base Directory | ![xdg](https://img.shields.io/github/labels/yuri-norwood/dotfiles/xdg) ![x11](https://img.shields.io/github/labels/yuri-norwood/dotfiles/x11) | `XDG` describes issues that relate *primarily* to conforming to the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html), e.g. moving `~/.wget-hsts` to `$XDG_CACHE_HOME`. |
| GIT | Git | ![git](https://img.shields.io/github/labels/yuri-norwood/dotfiles/git) | `GIT` describes issues that relate *primarily* to `git(1)` and it's usage, independent of any git hosting service, such as GitHub or Bitbucket. |
| HUB | GitHub | ![github](https://img.shields.io/github/labels/yuri-norwood/dotfiles/github) | `HUB` describes issues that relate *primarily* to GitHub and maintaining it's workflows and documents. |
| ENV | UNIX Shell Environment / CLI | ![shell](https://img.shields.io/github/labels/yuri-norwood/dotfiles/shell) ![ksh](https://img.shields.io/github/labels/yuri-norwood/dotfiles/ksh) | `ENV` describes issues that relate *primarily* to the shell environment, i.e. more conventional dotfiles, such as `.profile` or `.xintrc`. |
| COL | Colours and themes | NA | `COL` describes issues that relate *primarily* to simple aesthetic changes, such as changing the colourscheme for vim / nvim, a suckless tool, just xterm. NOTE: Adding *new* colourschemes or playing with them is a `VIM` issue, not `COL`. |
| EDT | EditorConfig | ![editorconfig](https://img.shields.io/github/labels/yuri-norwood/dotfiles/editorconfig) | `EDT` describes issues that relate *primarily* to `.editorconfig` files through out the repo. |
| IDE | Visual Studio | ![visual studio](https://img.shields.io/github/labels/yuri-norwood/dotfiles/visual%20studio) | `IDE` describes issues that relate *primarily* to an IDE, such as Visual Studio (the only one for which there are currently configs). |
